src/include/daos/pool_map.h

```diff
@@ -30,14 +31,15 @@
  * yaml file.
  */
 typedef enum pool_comp_type {
-	PO_COMP_TP_TARGET	= 0, /** reserved, hard-coded */
-	PO_COMP_TP_RANK		= 1, /** reserved, hard-coded */
-	PO_COMP_TP_MIN		= 2, /** first user-defined domain */
-	PO_COMP_TP_NODE		= 2, /** for test only */
-	PO_COMP_TP_GRP		= 3, /** group, commonly used for performance domain */
-	PO_COMP_TP_MAX		= 254, /** last user-defined domain */
-	PO_COMP_TP_ROOT		= 255,
-	PO_COMP_TP_END		= 256,
+	PO_COMP_TP_TARGET = 0,   /** reserved, hard-coded */
+	PO_COMP_TP_RANK   = 1,   /** reserved, hard-coded */
+	PO_COMP_TP_NODE   = 2,   /** reserved, hard-coded */
+	PO_COMP_TP_MIN    = 3,   /** first user-defined domain */
+	PO_COMP_TP_FAULT  = 3,   /** user-defined fault domain (if not node) */
```

Comment:
Should this be 4?

Reply:
No, the minimum user-defined level is now FAULT in this PR.